domain/model/state/controllerstate.go

```diff
@@ -1802,3 +1803,47 @@ AND    ot.type = $dbPermission.object_type
 	}
 	return nil
 }
+
+// GetAllActivatedModelsUUIDQuery returns a SQL statement that will return all activated models uuids.
+func (st *State) GetAllActivatedModelsUUIDQuery() string {
+	return "SELECT uuid from model WHERE activated = true"
+}
+
+// GetActivatedModelUUIDs returns the subset of model uuids from the supplied list that are activated.
+// If no models associated with the uuids are activated then an empty slice is returned.
+func (st *State) GetActivatedModelUUIDs(ctx context.Context, uuids []string) ([]coremodel.UUID, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	type modelUUID struct {
+		UUID coremodel.UUID `db:"uuid"`
+	}
+
+	stmt, err := st.Prepare(`
+SELECT uuid AS &modelUUID.uuid
+FROM   v_model
+WHERE uuid IN ($S[:])
+`, sqlair.S{}, modelUUID{})
```

Comment:
We don't use sqlair.S{}, define a []string type.

Reply:
Ignore this please. There is no such consensus or rule. This is just personal preference at this stage and we have plenty of examples of using this without issue.